File: leetcode/python/junechallengeday3.py
class Solution:
    def twoCitySchedCost(self, costs: List[List[int]]) -> int:
        half = len(costs)//2
        costs.sort(key=lambda x: x[0] - x[1])
        return sum(list(zip(*costs[:half]))[0]) + sum(list(zip(*costs[half:]))[1])

File: leetcode/c++/junechallengeday3.cpp
class Solution {
public:
    
    int twoCitySchedCost(vector<vector<int>>& v1) {
        sort(v1.begin(),v1.end(),[&](vector<int>  a,vector<int>b){
            return a[0]-a[1]>b[0]-b[1];
        }) ;
        int ans=0 ,n=v1.size();
        for(int i=0;i<n;i++){
            if (i<n/2){
                ans+=v1[i][1];
            }
            else{
                ans+=v1[i][0];
            }
        }
        return ans;
    }
};
